datastructures/linkedlist/cyclicallylinkedlist/cyclicallylinkedlist.go

```diff
@@ -0,0 +1,144 @@
+// Package cyclicallylinkedlist demonstration of Linked List Cycle in golang
+
+package cyclicallylinkedlist
+
+import "fmt"
+
+// Node of List.
+type ClNode struct {
+	val  interface{}
+	prev *ClNode
+	next *ClNode
+}
+
+// The Cycling list in this implementation is addressed
+// by means of the element located at the current position.
+type ClList struct {
+	size        int
+	currentItem *ClNode
+}
+
+// Create new list.
+func NewList() *ClList {
+	return &ClList{0, nil}
+}
+
+// Create new node.
+func NewNode(val interface{}) *ClNode {
+	return &ClNode{val, nil, nil}
+}
+
+// Inserting the first node is a special case. It will
+// point to itself. For other cases, the node will be added
+// to the end of the list. End of the list is prev field of
+// current item. Complexity O(1).
+func (cl *ClList) Add(val interface{}) {
+	n := NewNode(val)
+	cl.size++
+	if cl.currentItem == nil {
+		n.prev = n
+		n.next = n
+		cl.currentItem = n
+	} else {
+		n.prev = cl.currentItem.prev
+		n.next = cl.currentItem
+		cl.currentItem.prev.next = n
+		cl.currentItem.prev = n
+	}
+}
+
+// Rotate method is interesting for optimization.
+// Let us need to rotate the list by P places
```

Comment:
Sorry the documentation here doesn't make sense. Could you maybe check and add a reference to it. The sentence that I marked doesn't make sense to me - maybe elaborate what you mean!

Reply:
Do you mean what is written is incomprehensible, or that this information is redundant?

Reply:
Its incomprehensible to me. I didn't understand the first sentence at all.